kernel/src/x86_64/arch_api/paging.rs

```diff
@@ -65,17 +66,28 @@ fn construct_page_table_entry(data: PageTableEntry) -> u64 {
     result
 }
 
+unsafe fn get_page_table_entry_address(
+    pml4_index: usize,
+    pml3_index: usize,
+    pml2_index: usize,
+    pml1_index: usize,
+) -> *mut u64 {
+    let offset =
+        pml1_index + pml2_index * 512 + pml3_index * 512 * 512 + pml4_index * 512 * 512 * 512;
+    RECURSIVE_PAGE_TABLE_POINTER.add(offset)
+}
```

Comment:
This is a cool improvement. Will prevent future bugs by removing duplication that could be incorrectly modified in only one place.